### PR TITLE
[keras][loss] kullback_leibler_divergence and poisson

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?
add kullback_leibler_divergence and Poisson for feature parity with keras.

## How was this patch tested?
unit test

### CODE REVIEW

File: docs/docs/APIGuide/Losses.md [hunk 1]
```diff
@@ -1987,3 +1987,117 @@ loss = criterion.forward(input, target)
...
+```scala
+val criterion = KullbackLeiblerDivergenceCriterion()
+```
+**Python:**
+```python
+criterion = CosineProximityCriterion()
+```
+
+Computes the negative of the mean cosine proximity between predictions and targets.
+>>>>>>> upstream/master
```
Comment (wzhongyuan) [inline docs/docs/APIGuide/Losses.md]: conflict

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KullbackLeiblerDivergenceCriterion.scala [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+import com.intel.analytics.bigdl.nn.abstractnn.TensorCriterion
+import com.intel.analytics.bigdl.tensor.{DenseTensorApply, Tensor, TensorFunc2}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * This method is same as `kullback_leibler_divergence` loss in keras.
+ * It calculates:
```
Comment (wzhongyuan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KullbackLeiblerDivergenceCriterion.scala]: Put the formula to Python layer as well?

Comment (hhbyyh) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/KullbackLeiblerDivergenceCriterion.scala]: Thanks @wzhongyuan  I will make a pass for the code tomorrow and will let you know once this is ready for review.

**Comment by hhbyyh** [comment]
This is ready for review. Thanks

**Comment by wzhongyuan** [comment]
LGTM

**Comment by hhbyyh** [comment]
PR validation passed.
http://172.168.2.101:8080/job/BigDL-PR-V/385/

**Comment by hhbyyh** [comment]
cc @zhichao-li for a final check.

**Comment by hhbyyh** [comment]
merge conflict

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/job/BigDL-PR-V/449/ PR validation passed.
